commands/plugin_e2e_test.go

```diff
@@ -0,0 +1,145 @@
+
+		cmd := &cobra.Command{}
+		_ = pluginEnableCmdF(c, cmd, []string{pluginID})
+		//err = pluginEnableCmdF(c, cmd, []string{pluginID})
+		//s.Require().Error(err)
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 1)
+		s.Require().Equal(printer.GetErrorLines()[0], "Unable to enable plugin: "+pluginID+". Error: : You do not have the appropriate permissions., ")
+
+		plugins, appErr = s.th.App.GetPlugins()
+		s.Require().Nil(appErr)
+		s.Require().Len(plugins.Active, 0)
+		s.Require().Len(plugins.Inactive, 1)
+	})
+
+	s.RunForSystemAdminAndLocal("error for enabling non existent plugin", func(c client.Client) {
+		printer.Clean()
+
+		plugins, appErr := s.th.App.GetPlugins()
+		s.Require().Nil(appErr)
+		s.Require().Len(plugins.Active, 0)
+		s.Require().Len(plugins.Inactive, 1)
+
+		cmd := &cobra.Command{}
+		_ = pluginEnableCmdF(c, cmd, []string{nonExistentPluginID})
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 1)
+		s.Require().Equal(printer.GetErrorLines()[0], "Unable to enable plugin: "+nonExistentPluginID+". Error: : Plugin is not installed., ")
+
+		plugins, appErr = s.th.App.GetPlugins()
+		s.Require().Nil(appErr)
+		s.Require().Len(plugins.Active, 0)
+		s.Require().Len(plugins.Inactive, 1)
+	})
+
+	s.RunUnauthenticatedClient("error for enabling non existent plugin", func(c client.Client) {
+		printer.Clean()
+
+		plugins, appErr := s.th.App.GetPlugins()
+		s.Require().Nil(appErr)
+		s.Require().Len(plugins.Active, 0)
+		s.Require().Len(plugins.Inactive, 1)
+
+		cmd := &cobra.Command{}
+		_ = pluginEnableCmdF(c, cmd, []string{nonExistentPluginID})
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 1)
+		s.Require().Equal(printer.GetErrorLines()[0], "Unable to enable plugin: "+nonExistentPluginID+". Error: : You do not have the appropriate permissions., ")
+
+		plugins, appErr = s.th.App.GetPlugins()
+		s.Require().Nil(appErr)
+		s.Require().Len(plugins.Active, 0)
+		s.Require().Len(plugins.Inactive, 1)
+	})
+
+	s.RunForAllClients("error when plugins are disabled", func(c client.Client) {
+		printer.Clean()
+
+		s.th.App.UpdateConfig(func(cfg *model.Config) {
+			*cfg.PluginSettings.Enable = false
+		})
+
+		cmd := &cobra.Command{}
+		_ = pluginEnableCmdF(c, cmd, []string{pluginID})
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 1)
+		s.Require().Equal(printer.GetErrorLines()[0], "Unable to enable plugin: "+pluginID+". Error: : Plugins have been disabled. Please check your logs for details., ")
+	})
+}
+
+func installPlugin(s *MmctlE2ETestSuite, pluginID string, pluginURL string) {
+	s.th.App.UpdateConfig(func(cfg *model.Config) {
+		*cfg.PluginSettings.Enable = true
+	})
+
+	pluginFileBytes, err := s.th.App.DownloadFromURL(pluginURL)
+	s.Require().Nil(err)
+	_, err = s.th.App.InstallPlugin(bytes.NewReader(pluginFileBytes), true)
+	s.Require().Nil(err)
+}
```

Comment:
We don't need to download from internet, see #277 we can upload the test plugin.

Reply:
@isacikgoz I don't see pluginID from the #277
Is there a way to get it?

Reply:
I mean you can use `pluginURL` to open a file and supply it to `(*app.App).InstallPlugin(pluginFile io.ReadSeeker, replace bool)` since `os.File` satisfies the interface. Also since `pluginURL` is a file path I think it could be renamed as `testPluginFile`.